pkg/runtime/helper_test.go

```diff
@@ -63,6 +63,28 @@ func TestExtractList(t *testing.T) {
 	}
 }
 
+func TestExtractGenericList(t *testing.T) {
```

Comment:
@smarterclayton I still don't get what do you want to test :-(

The ExtractList accepts an runtime.Object, so you can use only api.*List{} or you need to make up some fake resource like:

```
type testObj struct {
	Items []*runtime.Object
}
func (t testObj) IsAnAPIObject() {}
```

Now, how you can now populate the Items with values when the `runtime.Object` is already an interface and you basically saying that you want to have a list of pointers to interfaces there (how that can happen?)

Can you please make an example or elaborate more? :-(